codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -191,11 +193,39 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         }
     }
 
+    private void onStreamIdResponse(Http2OutboundStreamIdResponse resp) {
+        assert isOutgoing(resp.streamId());
+
+        if (!(resp.cookie() instanceof Http2StreamChannel)) {
+            throw new IllegalArgumentException("cookie must be of type child channel.");
+        }
+        Http2StreamChannel childChannel = (Http2StreamChannel) resp.cookie();
+        childChannel.streamId(resp.streamId());
+        Http2StreamChannel existing = childChannels.put(resp.streamId(), childChannel);
+        assert existing == null;
+        // The channel might be closed already.
+        if (!childChannel.isOpen()) {
+            onStreamClosed(resp.streamId());
+        }
+        if (childChannel.isActive()) {
+            childChannel.pipeline().fireChannelActive();
+        }
+    }
+
     private void onStreamActive(int streamId) {
-        ChannelFuture future = createStreamChannel(ctx, streamId, streamHandler);
+        // Channels for local/outgoing streams are created directly via the Http2ChannelBootstrap and
+        // marked as active via onStreamIdResponse()
+        if (isOutgoing(streamId)) {
+            return;
+        }
+
+        Channel parentChannel = ctx.channel();
+        EventLoopGroup group = streamGroup != null ? streamGroup : parentChannel.eventLoop();
+
+        ChannelFuture future = createStreamChannel(parentChannel, group, streamHandler, streamId);
         Http2StreamChannel childChannel = (Http2StreamChannel) future.channel();
-        Http2StreamChannel oldChannel = childChannels.put(streamId, childChannel);
-        assert oldChannel == null;
+        Http2StreamChannel existing = childChannels.put(streamId, childChannel);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "existing". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)